logstash-core/src/main/java/org/logstash/common/BufferedTokenizerExt.java

```diff
@@ -106,12 +106,13 @@ public RubyArray extract(final ThreadContext context, IRubyObject data) {
     public IRubyObject flush(final ThreadContext context) {
         final IRubyObject buffer = input.join(context);
         input.clear();
+        inputSize = 0;
         return buffer;
     }
 
     @JRubyMethod(name = "empty?")
     public IRubyObject isEmpty(final ThreadContext context) {
-        return input.empty_p();
+        return RubyUtil.RUBY.newBoolean(input.isEmpty() && (inputSize == 0));
```

Comment:
I would _think_ it would be an illegal state for `(input.isEmpty()) != (inputSize == 0)`, but that isn't the case.

~~~
buftok.extract("this\nthat\nanother\n") #=> ["this","that","another"]
buftok.empty? #=> false
~~~

the `input` will effectively always have at least one entry after an `extract`, and zero entries after a flush, because in `extract` there are two situations:
 - no delimiters encountered, entries are added to `input`
 - delimiters encountered, input has exactly one entry (which is an empty string if the data was delimiter-terminated).

Reply:
from the comments, as I understand you're ok with the changes, can you throw in an actual approval on the PR?